packages/material/src/mui-controls/MuiAutocomplete.tsx

```diff
@@ -76,7 +76,7 @@ export const MuiAutocomplete = (props: EnumCellProps & WithClassname & WithOptio
       autoComplete
       fullWidth
       options={options}
-      getOptionLabel={getOptionLabel || (option => option?.label)}
+      getOptionLabel={getOptionLabel || (option => typeof option === 'string' ? option : option?.label)}
```

Comment:
The default value for getOptionLabel is `(option) => option.label ?? option` getOptionLabel. According to the docs the function provided has to accept string and the option type. There is a composed function of getOptionLabel and the anoymous one here. Because there is no guarantee getOptionLabel will return a valid value for any given string the TS compiler still thinks, this can be the union type string and the type of the options.

https://mui.com/material-ui/api/autocomplete/#:~:text=option%20to%20test.-,getOptionLabel,-func

Reply:
I'm using Intellij for development and get this error message on the label property:

> TS2339: Property 'label' does not exist on type 'string | EnumOption'.Property 'label' does not exist on type 'string'.

![Screenshot_20220613_120318_](https://user-images.githubusercontent.com/301227/173330599-4959edf0-5046-4709-84e7-738768da3b54.png)

Reply:
I was able to reproduce the problem by updating Material UI. I don't know why but for some reason the type inference fails here. I clicked trough Material UI's types and they seem to be correct (of course I maybe missed a detail somewhere). Still, although `freeSolo` is undefined it's treated as if it's `true`. As in `freeSolo` mode the option could be `string` this needs to be handled.

The type inference works correctly when explicitly handing over `false`.

```suggestion
      getOptionLabel={getOptionLabel || (option => option?.label)}
      freeSolo={false}
```